src/libs/antares/study/parts/hydro/series.h

```diff
@@ -153,25 +154,103 @@ class DataSeriesHydro
+    /**
+     * \brief Post processing numbers of TS's
+     *  Checking whether or not TS's numbers are different
+     *  and taking proper action based on corresponding check
+     */
+    bool postProcessMaxPowerTS(Area& area);
+
+    void setHydroModulability(Study& study, const AreaName& areaID) const;
+
+    /**
+     * \brief Setting TS's number of maximum generation and pumping matrices
+     */
+    void setNbTimeSeriesSup(uint nbTimeSeriesSup_);
+
+    /**
+     * \brief Setting TS's number of maximum generation and pumping matrices
+     */
+    void setNbTimeSeriesSup();
+
+    /**
+     * \brief Get function for TS's number of maximum generation and pumping matrices
+     */
+    uint getNbTimeSeriesSup() const;
+
+    /**
+     * \brief Setting TS's when derated mode is on
+     */
+    void setMaxPowerTSWhenDeratedMode(const Study& study);
+
+private:
     /*!
-    ** \brief The number of time-series for maxgen and maxpump matrices
+    ** \brief The number of time-series for maxHourlyGenPower and maxHourlyPumpPower matrices
     **
-    ** This value must be the same as the width of the matrices `maxgen` and `maxpump`.
+    ** This value must be the same as the width of the matrices `maxHourlyGenPower` and
+    *`maxHourlyPumpPower`.
     ** It is only provided for convenience to avoid same strange and ambiguous code
-    ** (for example using `maxgen.width` and `maxpump.width` in the same routine, it might
+    ** (for example using `maxHourlyGenPower.width` and `maxHourlyPumpPower.width` in the same
+    *routine, it might
     ** indicate that the two values are not strictly equal)
     */
-    uint countpowercredits = 0;
+    uint nbTimeSeriesSup = 0;
 
-    /*!
-    ** \brief Monte-Carlo
-    */
-    Matrix<uint32_t> timeseriesNumbers;
-    Matrix<uint32_t> timeseriesNumbersPowerCredits;
-    bool LoadHydroPowerCredits(Study& study, const AreaName& areaID, const AnyString& folder);
+    /**
+     * This class provides comparing two integers
+     * that represents width of two corresponding matrices.
+     */
+    class NbTsComparer
+    {
+    public:
+        NbTsComparer(uint32_t nbOfGenPowerTs_, uint32_t nbOfPumpPowerTs_);
+
+        bool bothZeros() const;
+        bool same() const;
+        bool differentAndGreaterThanOne(uint nbTimeSeriesSup_) const;
+
+    private:
+        uint32_t nbOfGenPowerTs{0};
+        uint32_t nbOfPumpPowerTs{0};
+    };
+
+    /**
+     * This class provides actions based on the return
+     * values of the NbTsComparer class member functions
+     */
+    class TsActions
+    {
+    public:
+        TsActions(Matrix<double, int32_t>& maxHourlyGenPower_,
+                  Matrix<double, int32_t>& maxHourlyPumpPower_);
+
+        void handleBothZeros(const AreaName& areaID);
+        [[noreturn]] void handleBothGreaterThanOne(const AreaName& areaID) const;
+        void resizeWhenOneTS(Area& area, uint nbTimeSeriesSup_);
+
+    private:
+        Matrix<double, int32_t>& maxHourlyGenPower;
```

Comment:
Please keep the trailing underscores for private variables, not for methods arguments.